ddtrace/tracer/textmap_test.go

```diff
@@ -637,6 +637,152 @@ func TestEnvVars(t *testing.T) {
 	}
 }
 
+func TestW3C(t *testing.T) {
+	t.Run("extract/valid", func(t *testing.T) {
+		var tests = []struct {
+			in              TextMapCarrier
+			traceID         uint64
+			spanID          uint64
+			priority        int
+			origin          string
+			propagatingTags map[string]string
+		}{
+			{
+				in: TextMapCarrier{
+					traceparentHeader: "00-00000000000000001111111111111111-2222222222222222-01",
+					tracestateHeader:  "dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
```

Comment:
Let's flip flop the order of the `tracestateHeader` in some of the tests just to be sure it'll still work (e.g. dd header comes after other vendors headers, the `o:` comes before the `s:`